### ISSUE TITLE
clang warning in elf_archive_tests

### ISSUE THREAD

**Issue by cipherboy** [rpminspect/rpminspect#37]
I'm not well-versed in `rpminspect`.

With clang turned up to 11 (`-Wall -Wextra -Werror`), I get the following output:

```C
inspect_elf.c:968:13: error: variable 'before_elf_fd' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
        if (after->peer_file != NULL) {
            ^~~~~~~~~~~~~~~~~~~~~~~~
inspect_elf.c:972:77: note: uninitialized use occurs here
        result = elf_archive_tests(ri, after_elf, after_elf_fd, before_elf, before_elf_fd, after->localpath, arch);
                                                                            ^~~~~~~~~~~~~
inspect_elf.c:968:9: note: remove the 'if' if its condition is always true
        if (after->peer_file != NULL) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
inspect_elf.c:948:22: note: initialize the variable 'before_elf_fd' to silence this warning
    int before_elf_fd;
                     ^
                      = 0
1 error generated.
```

The relevant snippet of code is [here](https://github.com/rpminspect/rpminspect/blob/master/src/librpminspect/inspect_elf.c#L966-L979):

```C
    /* Is this an archive or a regular ELF file? */
    if ((after_elf = get_elf_archive(after->fullpath, &after_elf_fd)) != NULL) {
        if (after->peer_file != NULL) {
            before_elf = get_elf_archive(after->peer_file->fullpath, &before_elf_fd);
        }

        result = elf_archive_tests(ri, after_elf, after_elf_fd, before_elf, before_elf_fd, after->localpath, arch);
    } else if ((after_elf = get_elf(after->fullpath, &after_elf_fd)) != NULL) {
        if (after->peer_file != NULL) {
            before_elf = get_elf_archive(after->peer_file->fullpath, &before_elf_fd);
        }

        result = elf_regular_tests(ri, after_elf, before_elf, after->localpath, arch);
    }
```

Looking at `elf_archive_tests`, it _looks_ like it assumes `after->peer_file != NULL` because it operates on both `before_elf` and `before_elf_fd` without checking their values and directly passes them to libelf. To me this seems like a bigger problem than the failure to initialize `before_elf_fd`. I could certainly initialize it to `-1`, but I'm worried that causes problems down the road.

Thoughts?

**Comment by dcantrell** [comment]
I'm not convinced clang is right here.  I get that it's freaking out about the uninitialized variable, but it isn't understanding how it's going to be used.  But even if it is read, it should give you zeros per C99 (and even C89?).

Nonetheless, I do not oppose initializing these to -1 to make ultra pedantic modes in C compilers less complainy.

**Comment by dcantrell** [comment]
So I fixed these things up anyway and rearranged some things to appease both gcc and clang.  I find it entirely reasonable just to keep the code base sane.  I tend to lean towards gcc but that's mostly due to the fact that I've used it forever (23 years by my count as of today).
